sunpy/net/dataretriever/sources/gong_synoptic.py

```diff
@@ -0,0 +1,157 @@
+                try:
+                    soup = BeautifulSoup(opn, "html.parser")
+                    for link in soup.find_all("a"):
+                        href = link.get("href")
+                        if re.match(file_pattern, href):
+                            # extracting timestamp from the filename
+                            timestring = href[5:16]
+                            tstamp = parse_time('20'+timestring.upper()+'00')
+                            if tstamp >= timerange.start and tstamp <= timerange.end:
+                                fullpath = directory + href
+                                filesurls.append(fullpath)
+                finally:
+                    opn.close()
+            except HTTPError as http_err:
+                if http_err.code == 404:
+                    continue
+                raise
+            except Exception:
+                raise
+        return filesurls
+
+    def fetch(self, qres, path=None, error_callback=None, **kwargs):
+        """
+        Download a set of results.
+
+        Parameters
+        ----------
+        qres : `~sunpy.net.dataretriever.QueryResponse`
+            Results to download.
+
+        path : `str`, optional
+            The directory to retrieve the files into.
+
+        Returns
+        -------
+        results: `parfive.Results`
+
+        """
+        if path is not None:
+            path = Path(path)
+
+        urls = [qrblock.url for qrblock in qres]
+
+        filenames = [url.split('/')[-1] for url in urls]
+        local_filenames = [name[:-3] for name in filenames]
+        paths = self._get_full_filenames(qres, filenames, path)
+        local_paths = self._get_full_filenames(qres, local_filenames, path)
+
+        downloader = Downloader()
+
+        for url, filename in zip(urls, paths):
+            downloader.enqueue_file(url, filename=filename)
+
+        resobj = downloader.download()
+
+        for loc_fpath, fpath in zip(local_paths, paths):
+            f = gzip.open(fpath, 'r')
+            g = gzip.open(loc_fpath, 'w')
+            g.write(f.read())
+
+        resobj.data = list(map(str, local_paths))
+        return resobj
+
+    def _makeimap(self):
+        """
+        Helper Function:used to hold information about source.
+        """
+        self.map_['instrument'] = 'GONG'
+        self.map_['source'] = 'NSO'
+
+    @classmethod
+    def _can_handle_query(cls, *query):
+        """
+        Answers whether client can service the query.
+
+        Parameters
+        ----------
+        query : list of query objects
+
+        Returns
+        -------
+        boolean
+            answer as to whether client can service the query
+        """
+        chkattr = ['Time', 'Instrument']
+        chklist = [x.__class__.__name__ in chkattr for x in query]
+        for x in query:
+            if x.__class__.__name__ == 'Instrument' and x.value.lower() == 'gong':
+                return all(chklist)
+        return False
```

Comment:
It is possible to download different types of data from GONG, so I think we need to put some more thought into making `_can_handle_query` more specific. I think what we might need is an attribute that specifies we are looking for a synoptic map (as opposed to full disk observation), and an attribute that specifies that we are looking for magnetic field (more specifically, radial magnetic field) data. I think I will open this in a new issue, so people can see/comment on ideas for these attributes (and whether we need to make any new ones).

Reply:
Thanks for the reivew, I have added physobs and extentType now.

Reply:
@dstansby this is green now, can be reviewed, I have done all suggested changes till now.